### PR TITLE
Redux setup

### PR DESCRIPTION
**Pull request by Hamza-777**
About this PR:
- Redux Setup.
- Added post create, delete functionalities.
- Added spinner.
- Added authentication.

### CODE REVIEW

File: src/Components/PostPage/PostPage.jsx [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+
+  if (loading) {
+    return <Spinner />;
+  }
+
   return (
     <section
-      class='container'
+      className='container'
       style={{ border: '2px solid', borderRadius: '7px' }}
```
Comment (Developers-codz) [inline src/Components/PostPage/PostPage.jsx:25]: Try to avoid inline styling

File: src/Components/Profile/Profile.jsx [hunk 2]
```diff
@@ -9,83 +9,107 @@ import {
...
 import { Link } from 'react-router-dom';
 import './Profile.css';
 
 const Profile = () => {
+  const { user } = useSelector((state) => state.auth);
   return (
     <section className='container'>
       <div className='myprofile'>
-        <FaTrashAlt className='icon delete-profile' />
+        {/* <FaTrashAlt className='icon delete-profile' /> */}
```
Comment (Developers-codz) [inline src/Components/Profile/Profile.jsx:22]: remove commented code if not needed

**Review by Developers-codz — APPROVED** [review]
code looks clean and readable, nothing much to suggest you can merge

File: src/Components/PostPage/PostPage.jsx (hunk 1 above)
Comment (Hamza-777) [reply src/Components/PostPage/PostPage.jsx:25]: Noted!

File: src/Components/Profile/Profile.jsx (hunk 2 above)
Comment (Hamza-777) [reply src/Components/Profile/Profile.jsx:22]: Will do it.

**Review by HrishikeshMahalle — APPROVED** [review]
The Code looks clean,just follow up peers suggestions, looks good to merge

**Pull request merged**